### PR TITLE
Dependabot updates 2

### PR DESCRIPTION
**Pull request by gmrabian**
### Description
More yarn fun for all the 🐈 out there

See comments for more detail

### Related ticket(s)

MDCT-2529

---
### How to test

App can start locally
passes checks
cypress still passes

---
### Author checklist

- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
---

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -99,7 +99,6 @@ jobs:
           else # We're not in a CI system, let's yarn install
             yarn install
           fi
-          cypress install
```
Comment (gmrabian) [inline .github/workflows/deploy.yml:102]: unecessary

File: package.json [hunk 2]
```diff
@@ -32,10 +32,6 @@
     "@types/node": "^14.14.6",
     "@types/yargs": "^15.0.10",
     "chromedriver": ">=88.0.0",
-    "cypress": "^8.5.0",
-    "cypress-cucumber-preprocessor": "^4.3.0",
-    "cypress-file-upload": "^5.0.8",
-    "cypress-xpath": "^1.6.2",
```
Comment (gmrabian) [inline package.json:38]: don't need cypress at top package, just tests/cypress folder

File: services/app-api/libs/dynamodb-lib.js [hunk 3]
```diff
@@ -56,6 +52,4 @@ export default {
   },
   batchWrite: (params) => client.batchWrite(params).promise(),
   listTables: (params) => database.listTables(params).promise(),
-  increment: (params) =>
-    atomicCounter.increment(params, { tableName: atomicTableName }),
```
Comment (gmrabian) [inline services/app-api/libs/dynamodb-lib.js:60]: not used anywhere, so removing this and the package

File: services/app-api/package.json [hunk 4]
```diff
@@ -27,8 +27,7 @@
     "serverless-stack-termination-protection": "^1.0.4"
   },
   "dependencies": {
-    "aws-sdk": "^2.891.0",
-    "dynamodb-atomic-counter": "^0.1.1",
```
Comment (gmrabian) [inline services/app-api/package.json:31]: atomic counter package not used

File: services/ui-src/serverless.yml [hunk 5]
```diff
@@ -6,8 +6,8 @@ plugins:
   - serverless-s3-bucket-helper
   - serverless-plugin-scripts
   - serverless-s3-sync
-  - serverless-cloudfront-invalidate
   - serverless-stack-termination-protection
+  - serverless-cloudfront-invalidate
```
Comment (gmrabian) [inline services/ui-src/serverless.yml:10]: docs instruct this to be last in plugin list

File: services/ui-src/src/components/Signup/Signup.js [hunk 6]
```diff
@@ -123,7 +123,7 @@ export default function Signup() {
           <ControlLabel>Last Name</ControlLabel>
           <FormControl
             autoFocus
-            type="lastName"
+            type="text"
```
Comment (gmrabian) [inline services/ui-src/src/components/Signup/Signup.js:126]: appropriate field types; also pages not used anyway

File: tests/cypress/plugins/index.js [hunk 7]
```diff
@@ -1,10 +1,31 @@
-module.exports = (on, config) => {
-  config.baseUrl = process.env.APPLICATION_ENDPOINT || "http://localhost:3000";
+const webpack = require("@cypress/webpack-preprocessor");
+const preprocessor = require("@badeball/cypress-cucumber-preprocessor");
```
Comment (gmrabian) [inline tests/cypress/plugins/index.js:2]: update plugins to work with webpack config

**Review by gmrabian — COMMENTED** [review]
highlighting a few changes here

cypress overview:
There was a single/double-nested thing going on with duplicate cypress configs and supporting files. I consolidated all that and also removed boilerplate spec files

File: .github/workflows/deploy.yml [hunk 8]
```diff
@@ -50,9 +50,11 @@ jobs:
       - uses: actions/setup-node@v1
         with:
           node-version: ${{ steps.node_version.outputs.NODE_VERSION }}
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         with:
-          path: "**/node_modules"
+          path: |
+            node_modules
+            services/**/node_modules
```
Comment (gmrabian) [inline .github/workflows/deploy.yml:57]: don't cache tests/cypress modules because they are installed in cypress step

**Review by gmrabian — COMMENTED** [review]
tested with cache miss and hit – ensured it works in both scenarios
[hit](https://github.com/Enterprise-CMCS/macpro-mdct-seds/actions/runs/6124752942/job/16625337846?pr=14724)
[miss](https://github.com/Enterprise-CMCS/macpro-mdct-seds/actions/runs/6124533840/job/16624736128)

**Review by BearHanded — APPROVED** [review]

**Pull request merged**